mantidproject/mantid
# Framework support for ISIS nexus log alarms

Adds additional logs alongside the affected log if invalid data is found.  Otherwise, nothing else changes.
If invalid data is found then the logs are automatically filtered just as for periods and running status.

**Report to:** @AnthonyLim23

**To test:**
1. Open workbench
1. Load ENGINX00228061_log_alarm_data.nxs from the unit test data directory
1. the log should get messages like this
   ```
   Load started
   Some "cryo_temp1" log data is invalid!
   Some "cryo_temp2" log data is invalid!
   Load successful, Duration 0.66 
   ```
1. right click ws->show sample logs
1. You should have a matching "invalid_values" log for:
   1. Cryo_temp1
   1. Cryo_temp2
   1. But NOT cryo_Sample
1. the values in the stats table should be prefiltered for the invalid values (the graph does not filter values).
   1. Cryo_temp1 should have 1 value of 7.0
   1. Cryo_temp2 should have 3 values of 3, 5 and 7 after filtering.
1.  run this in python and check the result
``` python 
ws=mtd["ENGINX00228061_log_alarm_data"]
run = ws.run()
log = run.getLogData("cryo_temp1")
print("should be 7",log.nthValue(0))
print("should be 3",log.unfiltered().nthValue(0))
```



Fixes #27767


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
